src/fu-engine.c

```diff
@@ -6937,6 +6937,13 @@ fu_engine_check_firmware_attributes(FuEngine *self, FuDevice *device)
 	subsystem = fu_udev_device_get_subsystem(FU_UDEV_DEVICE(device));
 	if (g_strcmp0(subsystem, "firmware-attributes") == 0) {
 		g_autoptr(GError) error = NULL;
+		if (add) {
+			FuBiosSettings *settings = fu_context_get_bios_settings(self->ctx);
+			if (fu_bios_settings_get_count(settings) > 0) {
```

Comment:
this seems like a single-use API call in the slow path -- can't we get the entire array and just check the length ourselves?